lib/controllers/issueish-detail-controller.js

```diff
@@ -56,44 +58,65 @@ export class BareIssueishDetailController extends React.Component {
     this.updateTitle();
   }
 
+  componentWillMount() {
```

Comment:
🚨 `componentWillMount` is deprecated and unsafe in new React code. Have a look at [the React docs](https://reactjs.org/docs/react-component.html#unsafe_componentwillmount) for an explanation.

There are alternative ways to derive state from props. Helpfully the React folks have [a blog post](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html) with some of the (modern) approaches to it. Elsewhere in the codebase right now we tend to use the "lengthy and bug-prone" approach with `static getDerivedStateFromProps`, but I have an open issue (#1611) to go through and upgrade it to one of their recommended alternatives.

Reply:
thanks for catching this!  That's really important.

I ended up using `getDerivedStateFromProps` because we're only memoizing to avoid gnarly nested lookups, not because we're trying to avoid recomputing something expensive so `memoize-one` didn't really make sense here.